cli/cmd/upgrade.go

```diff
@@ -325,7 +325,14 @@ func doUpgrade() error {
 	// if yes, they need to be installed separately as they have moved to their own charts
 	helmHelper := helm.NewHelper()
 
-	if err := helmHelper.UpgradeChart(keptnUpgradeChart, keptnReleaseName, keptnNamespace, nil); err != nil {
+	// fetch user-defined values of the previous Keptn installation and provide those to the upgrade command
+	// this will ensure that options such as 'apiGatewayNginx.type' will stay the same, but newly introduced values will correctly be set to their default
+	previousValues, err := helmHelper.GetValues(keptnReleaseName, keptnNamespace)
```

Comment:
do we have a unit test for this ? If no, it is possible to add it ?

Reply:
in both cases a bigger refactoring would be needed, so we decided to create a dedicated issue for that back then: https://github.com/keptn/keptn/issues/6439

Reply:
fine by me :)